runpod_worker/utils.py

```
@@ -0,0 +1,284 @@
+        image = Image.open(BytesIO(image_data))
+        
+        # Ensure directory exists
+        os.makedirs(os.path.dirname(output_path), exist_ok=True)
+        
+        # Save image
+        image.save(output_path)
+        
+        return output_path
+        
+    except Exception as e:
+        raise ValueError(f"Failed to save base64 image: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)